drivers/network/ndis/ndis/miniport.c

```diff
@@ -2249,6 +2249,15 @@ NdisIPnPStopDevice(
       Adapter->NdisMiniportBlock.AllocatedResourcesTranslated = NULL;
     }
 
+    if (Adapter->BusInterface)
+    {
+        if (Adapter->BusInterface->InterfaceDereference)
+            Adapter->BusInterface->InterfaceDereference(Adapter->BusInterface->Context);
+
+        ExFreePoolWithTag(Adapter->BusInterface, NDIS_TAG);
+        Adapter->BusInterface = NULL;
```

Comment:
I think you should set `BusInterfaceQueried` to false here, since the device can be re-started after stop